ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -265,10 +265,26 @@ MemberFunction::MemberFunction(HqlCppTranslator & _translator, BuildCtx & classc
 MemberFunction::MemberFunction(HqlCppTranslator & _translator, BuildCtx & classctx, const char * text, unsigned _flags) : translator(_translator), ctx(classctx), flags(_flags)
 {
     stmt = ctx.addQuotedFunction(text, (flags & MFdynamicproto) != 0);
+    if (flags & MFoptimize)
+        stmt->setOptimize(true);
+    else if (flags & MFnooptimize)
+        stmt->setOptimize(false);
```

Comment:
This looks like it's setting a boolean value (not 2 boolean values) and, so, it looks like optimize has in indeterminate value when neither MFoptimize and MFnooptimize is set.  I think it would be clearer to separate out the setter functions for 'optimize' and 'nooptimize'.

Reply:
I've renamed the function to hopefully make it clearer.  It is a 3 state - default/optimize/nooptimize and it will never be reset to the default.